File: Fundamentals-Java/MidExam26June/Numbers.js
function numbers(arr){
    let commandCount = 1
    let numArr = arr[0].split(" ")

    let commandLine = arr[commandCount]

    while (commandLine !== "Finish"){
        let command = commandLine.split(" ")[0]
        let value = commandLine.split(" ")[1]

        switch (command){
            case "Add":
                numArr.push(value)
                break
            case "Remove":
                for (let index = 0; index < numArr.length; index++) {
                    if (numArr[index] === value){
                        numArr.splice(index, 1)
                        break
                    }

                }
                break
            case "Replace":
                let replacement = commandLine.split(" ")[2]
                let toReplace = numArr.indexOf(value)
                if (numArr.includes(value)){
                    numArr.splice(toReplace, 1, replacement)
                }
                break
            case "Collapse":
                for (let index = 0; index < numArr.length; index++) {
                    if (numArr[index] < value){
                        numArr.splice(index, 1)
                    }
                }
                break
        }

        commandCount++
        commandLine = arr[commandCount]
    }
    console.log(numArr.join(" "))
}

numbers(["5 9 70 -56 9 9",
    "Replace 9 10",
    "Remove 9",
    "Finish"])

File: Fundamentals-Java/MidExam26June/Numbers.java
package MidExam26June;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class Numbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        List<Integer> numbers = Arrays.stream(scanner.nextLine().split(" "))
                .map(Integer::parseInt).collect(Collectors.toList());

        String commandLine = scanner.nextLine();

        while (!commandLine.equals("Finish")){
            String command = commandLine.split(" ")[0];
            int value = Integer.parseInt(commandLine.split(" ")[1]);

            switch (command){
                case "Add":
                    numbers.add(value);
                    break;
                case "Remove":
                    for (int index = 0; index < numbers.size(); index++) {
                        if (numbers.get(index) == value){
                            numbers.remove(index);
                            break;
                        }
                    }
                    break;
                case "Replace":
                    int replacement = Integer.parseInt(commandLine.split(" ")[2]);

                    if (numbers.contains(value)){
                        numbers.set(numbers.indexOf(value), replacement);
                    }
                    break;
                case "Collapse":
                    numbers.removeIf(integer -> integer < value);

                    break;
            }

            commandLine = scanner.nextLine();
        }
        System.out.println(numbers.toString().replaceAll("[\\[\\],]", ""));
    }
}
